homeassistant/components/powerwall/switch.py

```diff
@@ -0,0 +1,74 @@
+"""Support for Powerwall Switches (V2 API only)."""
+
+from typing import Any
+
+from tesla_powerwall import GridStatus, IslandMode, Powerwall
+
+from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN, POWERWALL_API, POWERWALL_COORDINATOR
+from .models import PowerwallRuntimeData
+
+ICON = "mdi:home-battery-outline"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Powerwall switch platform from Powerwall resources."""
+    powerwall_data: PowerwallRuntimeData = hass.data[DOMAIN][config_entry.entry_id]
+    power_wall = powerwall_data[POWERWALL_API]
+    coordinator = powerwall_data[POWERWALL_COORDINATOR]
+    assert coordinator is not None
+
+    async_add_entities([PowerwallOffGridEnabledEntity(power_wall, coordinator)])
+
+
+class PowerwallOffGridEnabledEntity(SwitchEntity):
```

Comment:
```suggestion
class PowerwallOffGridEnabledEntity(PowerWallEntity, SwitchEntity):
```

Reply:
Ahh excellent, thanks for the pointer.

I've updated this and also removed all the unnecessary coordinator and power_wall initialisation code from switch->async_setup_entry